src/axom/sidre/core/Group.cpp

```diff
@@ -1077,16 +1079,20 @@ Group* Group::createGroup(const std::string& path, bool is_list)
     return nullptr;
   }
 
-  Group* new_group = new(std::nothrow) Group(intpath, group->getDataStore(), is_list);
-  if(new_group == nullptr)
+  if(!group->hasGroup(intpath))
   {
-    return nullptr;
+    Group* new_group = new(std::nothrow) Group(intpath, group->getDataStore(), is_list);
```

Comment:
I realize this predates this PR, but we should probably have a comment about why we're using the nothrow variant here (but not elsewhere). Does anyone recall the reason?